### PR TITLE
2 finished use case interactor UserManagerInteractor.java

### PR DESCRIPTION
**Pull request by alexyin0429**
This UserManagerInteractor is a Use Case Interactor. It handles all the use cases that affect the user itself. For example, want to know user current VIP status. Use cases that contains interaction between users like show Users that this user has liked will be handled by another Interactor which is being developed by Zezhou Cui.

### CODE REVIEW

File: src/main/java/use_cases/UserManagerInteractor.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        this.user = user;
+    }
+
+    public List<User> viewLiked(){
+        return user.showLiked();
+    }
+
+    public boolean viewAccountStatus(){
+        return user.showVip();
+    }
```
Comment (czz0120) [inline src/main/java/use_cases/UserManagerInteractor.java:24]: Does viewAccountStatus need to show restriction information?

Comment (alexyin0429) [reply src/main/java/use_cases/UserManagerInteractor.java:25]: Yes, added separated methods to show restriction and vipstatus. Also added a method to return a string of both restriction status and VIP status

Comment (alexyin0429) [reply src/main/java/use_cases/UserManagerInteractor.java:25]: > Does viewAccountStatus need to show restriction information?

Yes, added separated methods to show restriction and vipstatus. Also added a method to return a string of both restriction status and VIP status

**Review by ryancys1234 — APPROVED** [review]
Looks good and clean!

File: src/main/java/entities/UserFactory.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    public User create(String password, String accountName, Profile profile, boolean isVip,
+                       List<String> liked, List<String> likedme, Hashtable<Integer, List<Object>> reviews){
+        return new VipUser(password, accountName, profile, true, liked, likedme, reviews);
+    }
+
+    public User create(String password, String accountName, Profile profile,
+                       List<String> liked, List<String> likedme, Hashtable<Integer, List<Object>> reviews){
+        return new RegularUser(password, accountName, profile, liked, likedme, reviews);
+    }
+
```
Comment (czz0120) [inline src/main/java/entities/UserFactory.java:16]: Conflict with the user factory of Yifei's User Register use case.

**Review by czz0120 — COMMENTED** [review]
some suggestions:
1. make UserManagerInteractor abstract
2. A controller class, its constructor uses the UserManagerInteractor as parameter, since it is an abstraction.  A method maybe named viewUserStatus, takes in a String of a user's account name, pass it to a method(according to what the user want to do) of the  interactor class.
3.  After The interactor received the account name, find the corresponding User instance, and make some changes.
4. Then the interactor "talks" to presenter, to present information to user.

**Review by Tristal25 — DISMISSED** [review]
Looks good for me

**Review by czz0120 — APPROVED** [review]

File: src/main/java/Main.java [hunk 3]
```diff
@@ -1,43 +1,43 @@
...
+//        // Build the GUI, plugging in the parts
+//        ReviewCreationScreen creationScreen = new ReviewCreationScreen(reviewController);
+//        screens.add(creationScreen, "welcome");
+//        cardLayout.show(screens, "create");
+//        application.pack();
+//        application.setVisible(true);
+//
+//
+//        ReviewDeletionScreen deletionScreen = new ReviewDeletionScreen(reviewController);
+//        screens.add(deletionScreen, "delete or hide");
```
Comment (czz0120) [inline src/main/java/Main.java:41]: Just out of curiosity,  why put all these commented codes into UserInfoScreen instead of  having them in  main.

Comment (alexyin0429) [reply src/main/java/Main.java:41]: They are in Main.java. They are just indented because I comment them out

**Review by czz0120 — APPROVED** [review]

**Review by czz0120 — DISMISSED** [review]

**Review by xumiche7 — DISMISSED** [review]
Looks good. Will link it to the search UI.

**Review by ryancys1234 — DISMISSED** [review]
Solid!

**Review by yifeiiZhang — DISMISSED** [review]

**Pull request merged**